SimTracker/TrackTriggerAssociation/interface/StubAssociation.h

```diff
@@ -0,0 +1,55 @@
+#ifndef SimTracker_TrackTriggerAssociation_StubAssociation_h
+#define SimTracker_TrackTriggerAssociation_StubAssociation_h
+
+#include "SimTracker/TrackTriggerAssociation/interface/TTTypes.h"
+#include "L1Trigger/TrackTrigger/interface/Setup.h"
+
+#include <vector>
+#include <map>
+
+namespace tt {
+
+  /*! \class  tt::StubAssociation
+   *  \brief  Class to store the MC truth of L1 Track Trigger stubs
+   *  \author Thomas Schuh
+   *  \date   2020, Apr
+   */
+  class StubAssociation {
+  public:
+    StubAssociation() { setup_ = nullptr; }
+    StubAssociation(const Setup* setup) : setup_(setup) {}
+    ~StubAssociation() {}
+
+    // insert a TPPtr and its associated collection of TTstubRefs into the underlayering maps
+    void insert(const TPPtr& tpPtr, const std::vector<TTStubRef>& ttSTubRefs);
+    // returns map containing TTStubRef and their associated collection of TPPtrs
+    const std::map<TTStubRef, std::vector<TPPtr>>& getTTStubToTrackingParticlesMap() const { return mapTTStubRefsTPPtrs_; }
+    // returns map containing TPPtr and their associated collection of TTStubRefs
+    const std::map<TPPtr, std::vector<TTStubRef>>& getTrackingParticleToTTStubsMap() const { return mapTPPtrsTTStubRefs_; }
+    // returns collection of TPPtrs associated to given TTstubRef
+    const std::vector<TPPtr>& findTrackingParticlePtrs(const TTStubRef& ttStubRef) const;
+    // returns collection of TTStubRefs associated to given TPPtr
+    const std::vector<TTStubRef>& findTTStubRefs(const TPPtr& tpPtr) const;
+    // total number of stubs associated with TPs
+    int numStubs() const { return mapTTStubRefsTPPtrs_.size(); };
+    // total number of TPs associated with stubs
+    int numTPs() const { return mapTPPtrsTTStubRefs_.size(); };
+    // returns all TPs which are matched treating stub collection as track
```

Comment:
Make comment more precise. e.g. "Get all TPs that are matched to these stubs in at least 'tpMinLayers' layers "?